### ISSUE TITLE
hash_bookmark doesn't work with non-ASCII headlines and blank headlines

### ISSUE THREAD

**Issue by JoeGermuska** [NUKnightLab/TimelineJS3#807]
We got a support request from the creator of a Chinese-language timeline who wanted to use hash bookmarks.

They were failing because the slide IDs were different every time the page is loaded.

When a `TimelineConfig` object is instantiated, populated, it attempts to ensure that all slides have unique identifiers, mostly to support the hash bookmark feature. If the configuration is JSON based, the IDs can be specified using 
the `unique_id` property of a [slide object](http://timeline.knightlab.com/docs/json-format.html#json-slide), but for most people using Spreadsheet configuration, there is no direct control over the IDs.

If no ID is specified, timeline [tries to construct one from the headline](https://github.com/NUKnightLab/TimelineJS3/blob/master/src/js/core/TimelineConfig.js/#L211-L227), but the `slugify` function filters out non-ASCII characters, leaving an empty "slug" for Chinese-language headlines (among others). 

That function calls out to another, [ensureUniqueKey](https://github.com/NUKnightLab/TimelineJS3/blob/master/src/js/core/Util.js#L320-L340) which is designed to do things like add numbers if the same headline is used more than once. If this function gets a blank ID, it generates a random six-character string. Since these have no relationship to the content of the slide, they will almost never come up the same way twice.

The tl;dr is that for timelines that have slides with non-ascii headlines or blank headlines, the hash_bookmark function will not work, unless the timelines are configured using JSON with explicit IDs specified.

Probably we need to come up with some kind of hashing function, which could be applied to non-ascii strings, but there would still be an issue with slides which have no headline. I'd probably be OK with saying that people who want to use hash_bookmark and have slides with blank headlines should use the JSON format, since it seems like quite an edge case.
